packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -383,6 +383,10 @@ class AndroidDevice extends Device {
       // activity name from the .apk.
       package = await AndroidApk.fromAndroidProject(project.android);
     }
+    // There was a failure parsing the android project information.
+    if (package == null) {
```

Comment:
What's the course of action for the user here?

Reply:
Address one of the error messages that was printed out from `await AndroidApk.fromAndroidProject(project.android)`

Reply:
Is there some way that we can mention that in the `throwToolExit` here?  Or some way we can assert that `AndroidApk.fromAndroidProject` reported some errors?  Perhaps have it `throwToolExit` with the errors instead of doing that here (maybe having an optional parameter like `{bool throwOnError = false}`?

Reply:
Throwing in the factory would change the contract in a way that I don't really want for now